### DISCUSSION TITLE
Infinite loop build in `development` when read, write have same parent dir

### DISCUSSION THREAD

**Discussion by mid-guy** [webpack/webpack#17891] [Q&A]
I got nfinite loop build in `development` when read, write have same parent dir in the plugin `GenerateLanguageJsonFilePlugin`
-> I found that if i read and write content in some file have same parent dir like ( 
read: path.join('src/locales', 'common.translation.json')
write: path.join('src/locales/en', 'common.translation.json')  
) -> INFINITE LOOP

I dont understand why :( 

webpack.common.js
`
/* eslint-disable @typescript-eslint/no-var-requires */
const path = require('path')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const CopyPlugin = require('copy-webpack-plugin')
const GenerateLanguageJsonFilePlugin = require('./plugins/GenerateLanguageJsonFilePlugin')

module.exports = {
  entry: path.resolve(__dirname, '..', './src/index.tsx'),
  resolve: {
    extensions: ['.tsx', '.ts', '.js'],
    alias: {
      '@components': path.resolve(__dirname, '../src/components'),
      // '@en': path.resolve(__dirname, 'translations.json'),
      // '@vn': path.resolve(__dirname, 'translations.json'),
    },
  },
  module: {
    rules: [
      {
        test: /\.(ts|js)x?$/,
        exclude: /node_modules/,
        use: [
          {
            loader: 'babel-loader',
          },
        ],
      },
      {
        test: /\.(scs|cs)s?$/,
        use: ['style-loader', 'css-loader'],
      },
      {
        test: /\.(?:ico|gif|png|jpg|jpeg)$/i,
        type: 'asset/resource',
      },
      {
        test: /\.(woff(2)?|eot|ttf|otf|svg|)$/,
        type: 'asset/inline',
      },
      {
        test: /\.json$/,
        type: 'asset/resource', // Use file-loader for JSON files
        use: 'json-loader',
        // include: path.resolve(__dirname, 'src/locales'), // Adjust the path to your JSON files
        // include: path.resolve(__dirname, 'locales'), // Chỉ xử lý tệp JSON trong thư mục locales
      },
    ],
  },
  output: {
    path: path.resolve(__dirname, '..', './build'),
    filename: '[name].js',
    clean: true
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: path.resolve(__dirname, '..', './src/index.html'),
    }),
    // process.env === "production" && new CopyPlugin({
    //   patterns: [
    //     { from: "src/locales", to: "build/" }
    //   ],
    // }),
    new GenerateLanguageJsonFilePlugin()
  ],
  // watchOptions: {
  //   ignored:
  //     [
  //       '**/node_modules',
  //       'src/locales/common/vn.json',
  //       'src/locales/common/en.json'
  //     ], 
  // },
  // watch: true,
}
`
`
/* eslint-disable @typescript-eslint/no-var-requires */
const write = require('write')
const path = require('path')
const fs = require('fs')

const { readdirSync, readFileSync, writeFileSync ,watchFile } = fs
const { join, basename } = path
const { sync } = write

class GenerateLanguageJsonFilePlugin {
  apply(compiler) {
    compiler.hooks.emit.tapAsync('GenerateLanguageJsonFilePlugin',  (compilation, done) => {
      // this._writeLanguageFiles()
      console.log('My plugin is running!')
      const data = JSON.parse(fs.readFileSync('src/locales/common.translation.json'))
      if ( !data ) {
        throw new Error('Check lai code')
      }
      const enData = { ...data }
      const vnData = { ...data }
      const enFilePath =. path.join('src/locales/en', 'common.translation.json')
      fs.writeFileSync(enFilePath, JSON.stringify(enData, null, 2), (err) => {
        if (err) {   
          console.log(new Error('Error creating the file:', err))
          // done()
        } else {
          console.log(`File 'en.json' has been created.`)
        }
      })
      done()
    })
  }
}

module.exports = GenerateLanguageJsonFilePlugin
`

**Comment by alexander-akait** [comment]
Please create reproducible test repo
